### PR TITLE
Use platform-apps with a branch

### PR DESCRIPTION
**Pull request by avnes**

### CODE REVIEW

File: _sub/compute/k8s-traefik-flux/dependencies.tf [hunk 1]
```diff
@@ -62,7 +62,7 @@ locals {
     "apiVersion" = "kustomize.config.k8s.io/v1beta1"
     "kind"       = "Kustomization"
     "resources" = [
-      "https://github.com/dfds/platform-apps/apps/${var.deploy_name}"
+      "${var.gitops_apps_repo_url}/apps/${var.deploy_name}?ref=${var.gitops_apps_repo_branch}"
```
Comment (emil2k) [inline _sub/compute/k8s-traefik-flux/dependencies.tf:65]: Shouldn't you run this through `urlencode`?

Comment (emil2k) [reply _sub/compute/k8s-traefik-flux/dependencies.tf:65]: Does a branch with a forward slash work?

File: _sub/compute/k8s-fluxcd/vars.tf [hunk 2]
```diff
@@ -23,6 +23,18 @@ variable "github_owner" {
   description = ""
 }
 
+variable "gitops_apps_repo_url" {
+  type        = string
+  default     = ""
```
Comment (emil2k) [inline _sub/compute/k8s-fluxcd/vars.tf:28]: Would it not be better if these didn't have a default value? This default does not seem valid.

**Review by emil2k — CHANGES_REQUESTED** [review]
Just some questions.

File: _sub/compute/k8s-traefik-flux/dependencies.tf (hunk 1 above)
Comment (avnes) [reply _sub/compute/k8s-traefik-flux/dependencies.tf:65]: Yes, it works, it has been tested in my sandbox. Here is an output from my sandbox:

`platform-apps-sources                     	feature/pss-psa@sha1:9d3ee061	False    	True 	Applied revision: feature/pss-psa@sha1:9d3ee061`

File: _sub/compute/k8s-fluxcd/vars.tf (hunk 2 above)
Comment (avnes) [reply _sub/compute/k8s-fluxcd/vars.tf:28]: These two vars have a default, but that is in the main module. The logic was that the gitops_apps_repo_url and gitops_apps_repo_branch could potentially be used by other main modules or by non-DFDS users.

**Review by emil2k — APPROVED** [review]

**Pull request merged**